R/yaml-as-df.R

```diff
@@ -53,10 +65,23 @@ yaml_as_df <- function(path, quiet = FALSE) { #nocov start
   if(rename) {
     names(ans) <- prototype_names
   }
+
   if(exists("cols", meta)) {
     cols <- meta[["cols"]]
-    ans <- select(ans, cols, tidyselect::everything())
+    ans <- select(ans, all_of(cols), everything())
   }
+
+  if(is.character(row_var)) {
+    row_var0 <- row_var
+    i <- 1
+    while(row_var %in% names(ans)) {
+      row_var <- paste0(row_var0, "_", i)
+      i <- i + 1
+    }
+    ans[, row_var] <- spec_names
+    ans <- select(ans, all_of(unique(c(row_var, names(ans)))))
```

Comment:
Okay as is, but some thoughts while reading along:

 * I wasn't able to come up with a case where the names and `row_var` wouldn't already be unique (given the upstream collision logic).

 * this `select` call is quite similar to the one on line 71, so as a reader it caused me to do a double take to see if there was any reason the approach on line 71 couldn't be used.  (I don't think there is.)

 * Two suggestions to simplify the "add row, rearrange row" into one step:

   ```
   ans <- tibble(!!row_var := spec_names, ans)
   ```

   or

   ```
   ans <- tibble::add_column(ans, !!row_var := spec_names, .before = 1)
   ```

Reply:
Thanks; went with the `tibble()` solution.